ui/src/app/subnets/views/SubnetDetails/SubnetSummary/SubnetSummaryForm/SubnetSummaryForm.tsx

```diff
@@ -25,80 +25,76 @@ const subnetSummaryFormSchema = Yup.object().shape({
   active_discovery: Yup.boolean(),
   allow_proxy: Yup.boolean(),
   allow_dns: Yup.boolean(),
-  vlan: Yup.number(),
+  vlan: Yup.number().required("VLAN is required"),
 });
 
+type Props = {
+  handleDismiss: () => void;
+  id: Subnet[SubnetMeta.PK];
+};
+
 const SubnetSummaryForm = ({
-  subnet,
   handleDismiss,
-}: {
-  subnet: Subnet;
-  handleDismiss: () => void;
-}): JSX.Element => {
+  id,
+}: Props): JSX.Element | null => {
+  const dispatch = useDispatch();
   const subnetErrors = useSelector(subnetSelectors.errors);
   const saving = useSelector(subnetSelectors.saving);
   const saved = useSelector(subnetSelectors.saved);
-  const dispatch = useDispatch();
-
+  const subnet = useSelector((state: RootState) =>
+    subnetSelectors.getById(state, id)
+  );
   const vlan = useSelector((state: RootState) =>
     vlanSelectors.getById(state, subnet?.vlan)
   );
 
   useEffect(() => {
-    dispatch(vlanActions.fetch());
     dispatch(fabricActions.fetch());
+    dispatch(vlanActions.fetch());
   }, [dispatch]);
 
+  if (!subnet || !vlan) {
+    return null;
+  }
```

Comment:
Instead of using a bunch of optional chaining in the form, I think it's cleaner to just exit out quick if we don't have the data we need. With this the TS compiler knows that beyond this conditional `subnet` and `vlan` definitely exist.

A better option would be to show a spinner but I took a shortcut here.